src/test/java/org/gridsuite/loadflow/server/LoadFlowParametersTest.java

```diff
@@ -140,28 +142,72 @@ void testCreateWithDefaultValues() throws Exception {
         assertThat(createdParameters).recursivelyEquals(defaultLoadFlowParameters);
     }
 
-    @Test
-    void testRead() throws Exception {
-
-        LoadFlowParametersInfos parametersToRead = buildParameters();
-
-        UUID parametersUuid = saveAndRetunId(parametersToRead);
-
-        MvcResult mvcResult = mockMvc.perform(get(URI_PARAMETERS_GET_PUT + parametersUuid))
+    protected void testCreateAndGet(LoadFlowParametersInfos parametersToRead, LoadFlowParametersInfos expectedParameters) throws Exception {
+        //create parameters
+        MvcResult mvcPostResult = mockMvc.perform(post("/" + VERSION + "/parameters")
+                        .content(mapper.writeValueAsString(parametersToRead))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpectAll(
+                        status().isOk(),
+                        content().contentType(MediaType.APPLICATION_JSON)
+                ).andReturn();
+
+        UUID createdParametersUuid = mapper.readValue(mvcPostResult.getResponse().getContentAsString(), UUID.class);
+        assertNotNull(createdParametersUuid);
+
+        // get created parameters
+        MvcResult mvcResult = mockMvc.perform(get(URI_PARAMETERS_GET_PUT + createdParametersUuid))
                 .andExpect(status().isOk()).andReturn();
         String resultAsString = mvcResult.getResponse().getContentAsString();
         LoadFlowParametersInfos receivedParameters = mapper.readValue(resultAsString, new TypeReference<>() {
         });
 
-        assertThat(receivedParameters).recursivelyEquals(parametersToRead);
+        assertThat(receivedParameters).recursivelyEquals(expectedParameters);
+    }
+
+    protected void testCreateAndGet(LoadFlowParametersInfos parametersToRead) throws Exception {
+        testCreateAndGet(parametersToRead, parametersToRead);
+    }
+
+    @Test
+    void testRead() throws Exception {
+        List<List<Double>> limitReductions = List.of(List.of(1.0, 0.9, 0.8, 0.7), List.of(1.0, 0.9, 0.8, 0.7));
+
+        // Get no limits with no provider
+        testCreateAndGet(buildCommonAndSpecificParameters().build());
+        // Get no limits with a provider other than 'OpenLoadFlow'
+        testCreateAndGet(buildCommonAndSpecificParameters().provider(PROVIDER).build());
+        testCreateAndGet(buildCommonAndSpecificParameters()
+                .provider(PROVIDER)
+                .limitReductions(limitReductionService.createLimitReductions(limitReductions))
+                .build(), buildCommonAndSpecificParameters()
+                .provider(PROVIDER)
+                .limitReductions(null)
+                .build());
+
+        // Get default limits with 'OpenLoadFlow' provider
+        String defaultProvider = limitReductionService.getProviders().iterator().next();
```

Comment:
Rename `defaultProvider` by `provider`
Default provider et provider for limits can be different